lib/routes/routePUT.js

```diff
@@ -17,17 +17,34 @@ const encryptionHeaders = [
 const validStatuses = ['Enabled', 'Suspended'];
 const validMfaDeletes = [undefined, 'Enabled', 'Disabled'];
 
+const MAX_POST_LENGTH = 1024 * 1024 * 0.5; // 512 KB
```

Comment:
Here's a little gimmick.

I noticed this PR in my inbox so came to take a look. Whilst looking at the changeset, the line above occurred to me as 'slightly odd', more exactly the choice to multiply by 0.5 instead of dividing by 2 (which admittedly is the same).

In some (statically typed, e.g. Haskell) languages, this wouldn't compile. In some other (statically typed, e.g. C) languages, this would work and the result type would be... depending on the value's type definition. In some (dynamically typed, e.g. Python) the result of the above would be 'float' whilst the type of the dividing version would be 'int'.

Now, in JavaScript all numeric values are of some 'float' type, so all of this shouldn't really make much of a difference, right?
V8 being an optimizing multi-staged JIT compiler, I was wondering whether it indeed doesn't impact the JIT's behaviour: at a language level it shouldn't, but a JIT could infer through runtime observation whether e.g. this value is at all times used with some value which is 'integer-like', and optimize accordingly.

Let's assume the `MAX_POST_LENGTH` value is used only once in the code, when compared to `postLength` which is at all times some kind of natural number, so the experiment below somewhat relates.

I created a build of V8 which permits to dump the assembly the JIT generates, and dump it again after every optimization. I fed it the following test script:

```js
const MAX_POST_LENGTH1 = 1024 * 1024 * 0.5;
const MAX_POST_LENGTH2 = 1024 * 1024 / 2;

function test1(a) {
    let result = 0;

    if(a > MAX_POST_LENGTH1) {
        result = 1;
    }
    else {
        result = 2;
    }

    return result;
}

function test2(a) {
    let result = 0;

    if(a > MAX_POST_LENGTH2) {
        result = 1;
    }
    else {
        result = 2;
    }

    return result;
}

while(1) {
    printErr(test1(128));
    printErr(test2(128));
}
```

When running this script, sending the output of `stderr` to `/dev/null` (the generated code is dumped on `stdout`, and we need some kind of side-effect for the calls to `test1` and `test2` not to be optimized away), we can see the optimizer kicking in a couple of times, starting with a bare-bones compilation of the whole unit, followed by a couple of optimizations of the two test functions, and ending up with a couple of whole-program inline/optimization rounds.

In my testing, whilst initially both versions get compiled into the same code (of 236 bytes), the second version (i.e. the one using an integer-like number) gets consistently optimized into a much simpler version (of 151 bytes) before the JIT decides to generate the same code for the first function.

As such, one may argue that using `* 0.5` is somewhat less efficient (at runtime) than the `/ 2` version.

This will obviously have 0 noticeable effect in an actual running server, the optimizer should kick in soon enough if `routePUT` is hot, so definitely not advising to change this line 😉

For reference, here's the whole output of V8 (3.29.86-18288-g87851fd) with generated code:

```
$ ./out/x64.release/d8 --print_opt_code --print-code test.js 2>/dev/null
--- Raw source ---
const MAX_POST_LENGTH1 = 1024 * 1024 * 0.5;
const MAX_POST_LENGTH2 = 1024 * 1024 / 2;

function test1(a) {
    let result = 0;

    if(a > MAX_POST_LENGTH1) {
        result = 1;
    }
    else {
        result = 2;
    }

    return result;
}

function test2(a) {
    let result = 0;

    if(a > MAX_POST_LENGTH2) {
        result = 1;
    }
    else {
        result = 2;
    }

    return result;
}

while(1) {
    printErr(test1(128));
    printErr(test2(128));
}


--- Code ---
source_position = 0
kind = FUNCTION
compiler = full-codegen
Instructions (size = 792)
0x21290d005820     0  55             push rbp
0x21290d005821     1  4889e5         REX.W movq rbp,rsp
0x21290d005824     4  56             push rsi
0x21290d005825     5  57             push rdi
0x21290d005826     6  488b4f2f       REX.W movq rcx,[rdi+0x2f]
0x21290d00582a    10  488b490f       REX.W movq rcx,[rcx+0xf]
0x21290d00582e    14  83411b01       addl [rcx+0x1b],0x1
0x21290d005832    18  41ff75a0       push [r13-0x60]
0x21290d005836    22  57             push rdi
0x21290d005837    23  49ba61e1d25e71100000 REX.W movq r10,0x10715ed2e161    ;; object: 0x10715ed2e161 <FixedArray[10]>
0x21290d005841    33  4152           push r10
0x21290d005843    35  b802000000     movl rax,0x2
0x21290d005848    40  48bba054b80000000000 REX.W movq rbx,0xb854a0
0x21290d005852    50  e869ebe7ff     call 0x21290ce843c0     ;; code: STUB, CEntryStub, minor: 8
0x21290d005857    55  488bf0         REX.W movq rsi,rax
0x21290d00585a    58  488945f8       REX.W movq [rbp-0x8],rax
0x21290d00585e    62  4d8b55a8       REX.W movq r10,[r13-0x58]
0x21290d005862    66  4c89562f       REX.W movq [rsi+0x2f],r10
0x21290d005866    70  4d8b55a8       REX.W movq r10,[r13-0x58]
0x21290d00586a    74  4c895637       REX.W movq [rsi+0x37],r10
0x21290d00586e    78  49ba71e3d25e71100000 REX.W movq r10,0x10715ed2e371    ;; object: 0x10715ed2e371 <FixedArray[6]>
0x21290d005878    88  4152           push r10
0x21290d00587a    90  6a00           push 0x0
0x21290d00587c    92  488b45f0       REX.W movq rax,[rbp-0x10]
0x21290d005880    96  488b402f       REX.W movq rax,[rax+0x2f]
0x21290d005884   100  488b400f       REX.W movq rax,[rax+0xf]
0x21290d005888   104  50             push rax
0x21290d005889   105  b803000000     movl rax,0x3
0x21290d00588e   110  48bb60f0b70000000000 REX.W movq rbx,0xb7f060
0x21290d005898   120  e823ebe7ff     call 0x21290ce843c0     ;; code: STUB, CEntryStub, minor: 8
0x21290d00589d   125  493ba5500d0000 REX.W cmpq rsp,[r13+0xd50]
0x21290d0058a4   132  7305           jnc 139  (0x21290d0058ab)
0x21290d0058a6   134  e855daf2ff     call StackCheck  (0x21290cf33300)    ;; code: BUILTIN
0x21290d0058ab   139  48b80000000056ff75f0 REX.W movq rax,0xf075ff5600000000
0x21290d0058b5   149  49ba0000000056ff7df0 REX.W movq r10,0xf07dff5600000000
0x21290d0058bf   159  4933c2         REX.W xorq rax,r10
0x21290d0058c2   162  488bce         REX.W movq rcx,rsi
0x21290d0058c5   165  4889412f       REX.W movq [rcx+0x2f],rax
0x21290d0058c9   169  488bd0         REX.W movq rdx,rax
0x21290d0058cc   172  f6c201         testb rdx,0x1
0x21290d0058cf   175  0f8426000000   jz 219  (0x21290d0058fb)
0x21290d0058d5   181  488d592f       REX.W leaq rbx,[rcx+0x2f]
0x21290d0058d9   185  4881e20000f8ff REX.W andq rdx,0xfffffffffff80000
0x21290d0058e0   192  f6420802       testb [rdx+0x8],0x2
0x21290d0058e4   196  7415           jz 219  (0x21290d0058fb)
0x21290d0058e6   198  48c7c20000f8ff REX.W movq rdx,0xfff80000
0x21290d0058ed   205  4823d1         REX.W andq rdx,rcx
0x21290d0058f0   208  f6420804       testb [rdx+0x8],0x4
0x21290d0058f4   212  7405           jz 219  (0x21290d0058fb)
0x21290d0058f6   214  e84542eaff     call 0x21290cea9b40     ;; code: STUB, RecordWriteStub, minor: 801
0x21290d0058fb   219  48b80000000056ff75f0 REX.W movq rax,0xf075ff5600000000
0x21290d005905   229  49ba0000000056ff7df0 REX.W movq r10,0xf07dff5600000000
0x21290d00590f   239  4933c2         REX.W xorq rax,r10
0x21290d005912   242  488bce         REX.W movq rcx,rsi
0x21290d005915   245  48894137       REX.W movq [rcx+0x37],rax
0x21290d005919   249  488bd0         REX.W movq rdx,rax
0x21290d00591c   252  f6c201         testb rdx,0x1
0x21290d00591f   255  0f8426000000   jz 299  (0x21290d00594b)
0x21290d005925   261  488d5937       REX.W leaq rbx,[rcx+0x37]
0x21290d005929   265  4881e20000f8ff REX.W andq rdx,0xfffffffffff80000
0x21290d005930   272  f6420802       testb [rdx+0x8],0x2
0x21290d005934   276  7415           jz 299  (0x21290d00594b)
0x21290d005936   278  48c7c20000f8ff REX.W movq rdx,0xfff80000
0x21290d00593d   285  4823d1         REX.W andq rdx,rcx
0x21290d005940   288  f6420804       testb [rdx+0x8],0x4
0x21290d005944   292  7405           jz 299  (0x21290d00594b)
0x21290d005946   294  e8f541eaff     call 0x21290cea9b40     ;; code: STUB, RecordWriteStub, minor: 801
0x21290d00594b   299  48b8112350e3ec240000 REX.W movq rax,0x24ece3502311    ;; object: 0x24ece3502311 <undefined>
0x21290d005955   309  488945e8       REX.W movq [rbp-0x18],rax
0x21290d005959   313  eb0f           jmp 330  (0x21290d00596a)
0x21290d00595b   315  493b45b8       REX.W cmpq rax,[r13-0x48]
0x21290d00595f   319  0f8405000000   jz 330  (0x21290d00596a)
0x21290d005965   325  e97a010000     jmp 708  (0x21290d005ae4)
0x21290d00596a   330  48b96177d25e71100000 REX.W movq rcx,0x10715ed27761    ;; object: 0x10715ed27761 <String[8]: printErr>
0x21290d005974   340  48b8000000000a000000 REX.W movq rax,0xa00000000
0x21290d00597e   350  e81d66f4ff     call LoadGlobalICTrampoline  (0x21290cf4bfa0)    ;; code: LOAD_GLOBAL_IC
0x21290d005983   355  488b75f8       REX.W movq rsi,[rbp-0x8]
0x21290d005987   359  50             push rax
0x21290d005988   360  49ba112350e3ec240000 REX.W movq r10,0x24ece3502311    ;; object: 0x24ece3502311 <undefined>
0x21290d005992   370  4152           push r10
0x21290d005994   372  48b941e0d25e71100000 REX.W movq rcx,0x10715ed2e041    ;; object: 0x10715ed2e041 <String[5]: test1>
0x21290d00599e   382  48b80000000002000000 REX.W movq rax,0x200000000
0x21290d0059a8   392  e8f365f4ff     call LoadGlobalICTrampoline  (0x21290cf4bfa0)    ;; code: LOAD_GLOBAL_IC
0x21290d0059ad   397  488b75f8       REX.W movq rsi,[rbp-0x8]
0x21290d0059b1   401  50             push rax
0x21290d0059b2   402  49ba112350e3ec240000 REX.W movq r10,0x24ece3502311    ;; object: 0x24ece3502311 <undefined>
0x21290d0059bc   412  4152           push r10
0x21290d0059be   414  49ba0000000080000000 REX.W movq r10,0x8000000000
0x21290d0059c8   424  4152           push r10
0x21290d0059ca   426  48ba000000000c000000 REX.W movq rdx,0xc00000000
0x21290d0059d4   436  488b7c2410     REX.W movq rdi,[rsp+0x10]
0x21290d0059d9   441  b801000000     movl rax,0x1
0x21290d0059de   446  e89dfdffff     call 0x21290d005780     ;; code: CALL_IC
0x21290d0059e3   451  488b75f8       REX.W movq rsi,[rbp-0x8]
0x21290d0059e7   455  48890424       REX.W movq [rsp],rax
0x21290d0059eb   459  48ba0000000008000000 REX.W movq rdx,0x800000000
0x21290d0059f5   469  488b7c2410     REX.W movq rdi,[rsp+0x10]
0x21290d0059fa   474  b801000000     movl rax,0x1
0x21290d0059ff   479  e87cfdffff     call 0x21290d005780     ;; code: CALL_IC
0x21290d005a04   484  488b75f8       REX.W movq rsi,[rbp-0x8]
0x21290d005a08   488  4883c408       REX.W addq rsp,0x8
0x21290d005a0c   492  48b96177d25e71100000 REX.W movq rcx,0x10715ed27761    ;; object: 0x10715ed27761 <String[8]: printErr>
0x21290d005a16   502  48b8000000000a000000 REX.W movq rax,0xa00000000
0x21290d005a20   512  e87b65f4ff     call LoadGlobalICTrampoline  (0x21290cf4bfa0)    ;; code: LOAD_GLOBAL_IC
0x21290d005a25   517  488b75f8       REX.W movq rsi,[rbp-0x8]
0x21290d005a29   521  50             push rax
0x21290d005a2a   522  49ba112350e3ec240000 REX.W movq r10,0x24ece3502311    ;; object: 0x24ece3502311 <undefined>
0x21290d005a34   532  4152           push r10
0x21290d005a36   534  48b961e0d25e71100000 REX.W movq rcx,0x10715ed2e061    ;; object: 0x10715ed2e061 <String[5]: test2>
0x21290d005a40   544  48b80000000005000000 REX.W movq rax,0x500000000
0x21290d005a4a   554  e85165f4ff     call LoadGlobalICTrampoline  (0x21290cf4bfa0)    ;; code: LOAD_GLOBAL_IC
0x21290d005a4f   559  488b75f8       REX.W movq rsi,[rbp-0x8]
0x21290d005a53   563  50             push rax
0x21290d005a54   564  49ba112350e3ec240000 REX.W movq r10,0x24ece3502311    ;; object: 0x24ece3502311 <undefined>
0x21290d005a5e   574  4152           push r10
0x21290d005a60   576  49ba0000000080000000 REX.W movq r10,0x8000000000
0x21290d005a6a   586  4152           push r10
0x21290d005a6c   588  48ba0000000010000000 REX.W movq rdx,0x1000000000
0x21290d005a76   598  488b7c2410     REX.W movq rdi,[rsp+0x10]
0x21290d005a7b   603  b801000000     movl rax,0x1
0x21290d005a80   608  e8fbfcffff     call 0x21290d005780     ;; code: CALL_IC
0x21290d005a85   613  488b75f8       REX.W movq rsi,[rbp-0x8]
0x21290d005a89   617  48890424       REX.W movq [rsp],rax
0x21290d005a8d   621  48ba000000000e000000 REX.W movq rdx,0xe00000000
0x21290d005a97   631  488b7c2410     REX.W movq rdi,[rsp+0x10]
0x21290d005a9c   636  b801000000     movl rax,0x1
0x21290d005aa1   641  e8dafcffff     call 0x21290d005780     ;; code: CALL_IC
0x21290d005aa6   646  488b75f8       REX.W movq rsi,[rbp-0x8]
0x21290d005aaa   650  4883c408       REX.W addq rsp,0x8
0x21290d005aae   654  488945e8       REX.W movq [rbp-0x18],rax
0x21290d005ab2   658  48bbc1e1d25e71100000 REX.W movq rbx,0x10715ed2e1c1    ;; object: 0x10715ed2e1c1 Cell for 6144
0x21290d005abc   668  83430bfe       addl [rbx+0xb],0xfe
0x21290d005ac0   672  791d           jns 703  (0x21290d005adf)
0x21290d005ac2   674  e899d7f2ff     call InterruptCheck  (0x21290cf33260)    ;; code: BUILTIN
0x21290d005ac7   679  48bbc1e1d25e71100000 REX.W movq rbx,0x10715ed2e1c1    ;; object: 0x10715ed2e1c1 Cell for 6144
0x21290d005ad1   689  49ba0000000000180000 REX.W movq r10,0x180000000000
0x21290d005adb   699  4c895307       REX.W movq [rbx+0x7],r10
0x21290d005adf   703  e975feffff     jmp 313  (0x21290d005959)
0x21290d005ae4   708  488b45e8       REX.W movq rax,[rbp-0x18]
0x21290d005ae8   712  48bbc1e1d25e71100000 REX.W movq rbx,0x10715ed2e1c1    ;; object: 0x10715ed2e1c1 Cell for 6144
0x21290d005af2   722  83430bfc       addl [rbx+0xb],0xfc
0x21290d005af6   726  791f           jns 759  (0x21290d005b17)
0x21290d005af8   728  50             push rax
0x21290d005af9   729  e862d7f2ff     call InterruptCheck  (0x21290cf33260)    ;; code: BUILTIN
0x21290d005afe   734  58             pop rax
0x21290d005aff   735  48bbc1e1d25e71100000 REX.W movq rbx,0x10715ed2e1c1    ;; object: 0x10715ed2e1c1 Cell for 6144
0x21290d005b09   745  49ba0000000000180000 REX.W movq r10,0x180000000000
0x21290d005b13   755  4c895307       REX.W movq [rbx+0x7],r10
0x21290d005b17   759  c9             leavel
0x21290d005b18   760  c20800         ret 0x8
0x21290d005b1b   763  498b45a0       REX.W movq rax,[r13-0x60]
0x21290d005b1f   767  e9c4ffffff     jmp 712  (0x21290d005ae8)
0x21290d005b24   772  0f1f4000       nop

Source positions:
 pc offset  position
         0         0
       139        37  statement
       162        37
       219        81  statement
       242        81
       313       409  statement
       330       418  statement
       330       418
       372       427
       426       427
       459       418
       492       444  statement
       492       444
       534       453
       588       453
       621       444
       759       467  statement

Deoptimization Output Data (deopt points = 0)

Back edges (size = 1)
ast_id  pc_offset  loop_depth
    62        679           1

0x10715ed2e421: [TypeFeedbackInfo]
 - ic_total_count: 4, ic_with_type_info_count: 0, ic_generic_count: 0

RelocInfo (size = 34)
0x21290d005839  embedded object  (0x10715ed2e161 <FixedArray[10]>)
0x21290d005853  code target (STUB)  (0x21290ce843c0)
0x21290d005870  embedded object  (0x10715ed2e371 <FixedArray[6]>)
0x21290d005899  code target (STUB)  (0x21290ce843c0)
0x21290d0058a7  code target (BUILTIN)  (0x21290cf33300)
0x21290d0058f7  code target (STUB)  (0x21290cea9b40)
0x21290d005947  code target (STUB)  (0x21290cea9b40)
0x21290d00594d  embedded object  (0x24ece3502311 <undefined>)
0x21290d00596c  embedded object  (0x10715ed27761 <String[8]: printErr>)
0x21290d00597f  code target (LOAD_GLOBAL_IC)  (0x21290cf4bfa0)
0x21290d00598a  embedded object  (0x24ece3502311 <undefined>)
0x21290d005996  embedded object  (0x10715ed2e041 <String[5]: test1>)
0x21290d0059a9  code target (LOAD_GLOBAL_IC)  (0x21290cf4bfa0)
0x21290d0059b4  embedded object  (0x24ece3502311 <undefined>)
0x21290d0059df  code target (CALL_IC)  (0x21290d005780)
0x21290d005a00  code target (CALL_IC)  (0x21290d005780)
0x21290d005a0e  embedded object  (0x10715ed27761 <String[8]: printErr>)
0x21290d005a21  code target (LOAD_GLOBAL_IC)  (0x21290cf4bfa0)
0x21290d005a2c  embedded object  (0x24ece3502311 <undefined>)
0x21290d005a38  embedded object  (0x10715ed2e061 <String[5]: test2>)
0x21290d005a4b  code target (LOAD_GLOBAL_IC)  (0x21290cf4bfa0)
0x21290d005a56  embedded object  (0x24ece3502311 <undefined>)
0x21290d005a81  code target (CALL_IC)  (0x21290d005780)
0x21290d005aa2  code target (CALL_IC)  (0x21290d005780)
0x21290d005ab4  embedded object  (0x10715ed2e1c1 Cell for 6144)
0x21290d005ac3  code target (BUILTIN)  (0x21290cf33260)
0x21290d005ac9  embedded object  (0x10715ed2e1c1 Cell for 6144)
0x21290d005aea  embedded object  (0x10715ed2e1c1 Cell for 6144)
0x21290d005afa  code target (BUILTIN)  (0x21290cf33260)
0x21290d005b01  embedded object  (0x10715ed2e1c1 Cell for 6144)

--- End code ---
--- Raw source ---
(a) {
    let result = 0;

    if(a > MAX_POST_LENGTH1) {
        result = 1;
    }
    else {
        result = 2;
    }

    return result;
}


--- Code ---
source_position = 101
kind = FUNCTION
name = test1
compiler = full-codegen
Instructions (size = 236)
0x21290d005bc0     0  55             push rbp
0x21290d005bc1     1  4889e5         REX.W movq rbp,rsp
0x21290d005bc4     4  56             push rsi
0x21290d005bc5     5  57             push rdi
0x21290d005bc6     6  488b4f2f       REX.W movq rcx,[rdi+0x2f]
0x21290d005bca    10  488b490f       REX.W movq rcx,[rcx+0xf]
0x21290d005bce    14  83411b01       addl [rcx+0x1b],0x1
0x21290d005bd2    18  41ff75a0       push [r13-0x60]
0x21290d005bd6    22  4d8b55a8       REX.W movq r10,[r13-0x58]
0x21290d005bda    26  4c8955e8       REX.W movq [rbp-0x18],r10
0x21290d005bde    30  493ba5500d0000 REX.W cmpq rsp,[r13+0xd50]
0x21290d005be5    37  7305           jnc 44  (0x21290d005bec)
0x21290d005be7    39  e814d7f2ff     call StackCheck  (0x21290cf33300)    ;; code: BUILTIN
0x21290d005bec    44  33c0           xorl rax,rax
0x21290d005bee    46  488945e8       REX.W movq [rbp-0x18],rax
0x21290d005bf2    50  ff7510         push [rbp+0x10]
0x21290d005bf5    53  488bc6         REX.W movq rax,rsi
0x21290d005bf8    56  488b402f       REX.W movq rax,[rax+0x2f]
0x21290d005bfc    60  493b45a8       REX.W cmpq rax,[r13-0x58]
0x21290d005c00    64  7520           jnz 98  (0x21290d005c22)
0x21290d005c02    66  49baf1dfd25e71100000 REX.W movq r10,0x10715ed2dff1    ;; object: 0x10715ed2dff1 <String[16]: MAX_POST_LENGTH1>
0x21290d005c0c    76  4152           push r10
0x21290d005c0e    78  b801000000     movl rax,0x1
0x21290d005c13    83  48bbf04bb50000000000 REX.W movq rbx,0xb54bf0
0x21290d005c1d    93  e89ee7e7ff     call 0x21290ce843c0     ;; code: STUB, CEntryStub, minor: 8
0x21290d005c22    98  5a             pop rdx
0x21290d005c23    99  e8d808eaff     call 0x21290cea6500     ;; code: COMPARE_IC UNINITIALIZED (id = 26)
0x21290d005c28   104  90             nop
0x21290d005c29   105  eb0f           jmp 122  (0x21290d005c3a)
0x21290d005c2b   107  493b45b8       REX.W cmpq rax,[r13-0x48]
0x21290d005c2f   111  0f840e000000   jz 131  (0x21290d005c43)
0x21290d005c35   117  e91c000000     jmp 150  (0x21290d005c56)
0x21290d005c3a   122  4885c0         REX.W testq rax,rax
0x21290d005c3d   125  0f8e13000000   jle 150  (0x21290d005c56)
0x21290d005c43   131  48b80000000001000000 REX.W movq rax,0x100000000
0x21290d005c4d   141  488945e8       REX.W movq [rbp-0x18],rax
0x21290d005c51   145  e90e000000     jmp 164  (0x21290d005c64)
0x21290d005c56   150  48b80000000002000000 REX.W movq rax,0x200000000
0x21290d005c60   160  488945e8       REX.W movq [rbp-0x18],rax
0x21290d005c64   164  488b45e8       REX.W movq rax,[rbp-0x18]
0x21290d005c68   168  48bbf1e7d25e71100000 REX.W movq rbx,0x10715ed2e7f1    ;; object: 0x10715ed2e7f1 Cell for 6144
0x21290d005c72   178  83430bd1       addl [rbx+0xb],0xd1
0x21290d005c76   182  791f           jns 215  (0x21290d005c97)
0x21290d005c78   184  50             push rax
0x21290d005c79   185  e8e2d5f2ff     call InterruptCheck  (0x21290cf33260)    ;; code: BUILTIN
0x21290d005c7e   190  58             pop rax
0x21290d005c7f   191  48bbf1e7d25e71100000 REX.W movq rbx,0x10715ed2e7f1    ;; object: 0x10715ed2e7f1 Cell for 6144
0x21290d005c89   201  49ba0000000000180000 REX.W movq r10,0x180000000000
0x21290d005c93   211  4c895307       REX.W movq [rbx+0x7],r10
0x21290d005c97   215  c9             leavel
0x21290d005c98   216  c21000         ret 0x10
0x21290d005c9b   219  498b45a0       REX.W movq rax,[r13-0x60]
0x21290d005c9f   223  e9c4ffffff     jmp 168  (0x21290d005c68)
0x21290d005ca4   228  0f1f4000       nop

Source positions:
 pc offset  position
         0       101
        44       124  statement
        46       124
        50       132  statement
        50       135
        53       139
        98       137
       131       167  statement
       141       174
       150       204  statement
       160       211
       164       227  statement
       164       234
       215       242  statement

Deoptimization Output Data (deopt points = 0)

Back edges (size = 0)
ast_id  pc_offset  loop_depth

0x10715ed2e821: [TypeFeedbackInfo]
 - ic_total_count: 1, ic_with_type_info_count: 0, ic_generic_count: 0

RelocInfo (size = 10)
0x21290d005be8  code target (BUILTIN)  (0x21290cf33300)
0x21290d005c04  embedded object  (0x10715ed2dff1 <String[16]: MAX_POST_LENGTH1>)
0x21290d005c1e  code target (STUB)  (0x21290ce843c0)
0x21290d005c24  code target with id (COMPARE_IC)  (0x21290cea6500) (id=26)
0x21290d005c6a  embedded object  (0x10715ed2e7f1 Cell for 6144)
0x21290d005c7a  code target (BUILTIN)  (0x21290cf33260)
0x21290d005c81  embedded object  (0x10715ed2e7f1 Cell for 6144)

--- End code ---
--- Raw source ---
(a) {
    let result = 0;

    if(a > MAX_POST_LENGTH2) {
        result = 1;
    }
    else {
        result = 2;
    }

    return result;
}


--- Code ---
source_position = 259
kind = FUNCTION
name = test2
compiler = full-codegen
Instructions (size = 236)
0x21290d005d40     0  55             push rbp
0x21290d005d41     1  4889e5         REX.W movq rbp,rsp
0x21290d005d44     4  56             push rsi
0x21290d005d45     5  57             push rdi
0x21290d005d46     6  488b4f2f       REX.W movq rcx,[rdi+0x2f]
0x21290d005d4a    10  488b490f       REX.W movq rcx,[rcx+0xf]
0x21290d005d4e    14  83411b01       addl [rcx+0x1b],0x1
0x21290d005d52    18  41ff75a0       push [r13-0x60]
0x21290d005d56    22  4d8b55a8       REX.W movq r10,[r13-0x58]
0x21290d005d5a    26  4c8955e8       REX.W movq [rbp-0x18],r10
0x21290d005d5e    30  493ba5500d0000 REX.W cmpq rsp,[r13+0xd50]
0x21290d005d65    37  7305           jnc 44  (0x21290d005d6c)
0x21290d005d67    39  e894d5f2ff     call StackCheck  (0x21290cf33300)    ;; code: BUILTIN
0x21290d005d6c    44  33c0           xorl rax,rax
0x21290d005d6e    46  488945e8       REX.W movq [rbp-0x18],rax
0x21290d005d72    50  ff7510         push [rbp+0x10]
0x21290d005d75    53  488bc6         REX.W movq rax,rsi
0x21290d005d78    56  488b4037       REX.W movq rax,[rax+0x37]
0x21290d005d7c    60  493b45a8       REX.W cmpq rax,[r13-0x58]
0x21290d005d80    64  7520           jnz 98  (0x21290d005da2)
0x21290d005d82    66  49ba19e0d25e71100000 REX.W movq r10,0x10715ed2e019    ;; object: 0x10715ed2e019 <String[16]: MAX_POST_LENGTH2>
0x21290d005d8c    76  4152           push r10
0x21290d005d8e    78  b801000000     movl rax,0x1
0x21290d005d93    83  48bbf04bb50000000000 REX.W movq rbx,0xb54bf0
0x21290d005d9d    93  e81ee6e7ff     call 0x21290ce843c0     ;; code: STUB, CEntryStub, minor: 8
0x21290d005da2    98  5a             pop rdx
0x21290d005da3    99  e85807eaff     call 0x21290cea6500     ;; code: COMPARE_IC UNINITIALIZED (id = 26)
0x21290d005da8   104  90             nop
0x21290d005da9   105  eb0f           jmp 122  (0x21290d005dba)
0x21290d005dab   107  493b45b8       REX.W cmpq rax,[r13-0x48]
0x21290d005daf   111  0f840e000000   jz 131  (0x21290d005dc3)
0x21290d005db5   117  e91c000000     jmp 150  (0x21290d005dd6)
0x21290d005dba   122  4885c0         REX.W testq rax,rax
0x21290d005dbd   125  0f8e13000000   jle 150  (0x21290d005dd6)
0x21290d005dc3   131  48b80000000001000000 REX.W movq rax,0x100000000
0x21290d005dcd   141  488945e8       REX.W movq [rbp-0x18],rax
0x21290d005dd1   145  e90e000000     jmp 164  (0x21290d005de4)
0x21290d005dd6   150  48b80000000002000000 REX.W movq rax,0x200000000
0x21290d005de0   160  488945e8       REX.W movq [rbp-0x18],rax
0x21290d005de4   164  488b45e8       REX.W movq rax,[rbp-0x18]
0x21290d005de8   168  48bbb9e9d25e71100000 REX.W movq rbx,0x10715ed2e9b9    ;; object: 0x10715ed2e9b9 Cell for 6144
0x21290d005df2   178  83430bd1       addl [rbx+0xb],0xd1
0x21290d005df6   182  791f           jns 215  (0x21290d005e17)
0x21290d005df8   184  50             push rax
0x21290d005df9   185  e862d4f2ff     call InterruptCheck  (0x21290cf33260)    ;; code: BUILTIN
0x21290d005dfe   190  58             pop rax
0x21290d005dff   191  48bbb9e9d25e71100000 REX.W movq rbx,0x10715ed2e9b9    ;; object: 0x10715ed2e9b9 Cell for 6144
0x21290d005e09   201  49ba0000000000180000 REX.W movq r10,0x180000000000
0x21290d005e13   211  4c895307       REX.W movq [rbx+0x7],r10
0x21290d005e17   215  c9             leavel
0x21290d005e18   216  c21000         ret 0x10
0x21290d005e1b   219  498b45a0       REX.W movq rax,[r13-0x60]
0x21290d005e1f   223  e9c4ffffff     jmp 168  (0x21290d005de8)
0x21290d005e24   228  0f1f4000       nop

Source positions:
 pc offset  position
         0       259
        44       282  statement
        46       282
        50       290  statement
        50       293
        53       297
        98       295
       131       325  statement
       141       332
       150       362  statement
       160       369
       164       385  statement
       164       392
       215       400  statement

Deoptimization Output Data (deopt points = 0)

Back edges (size = 0)
ast_id  pc_offset  loop_depth

0x10715ed2e9e9: [TypeFeedbackInfo]
 - ic_total_count: 1, ic_with_type_info_count: 0, ic_generic_count: 0

RelocInfo (size = 10)
0x21290d005d68  code target (BUILTIN)  (0x21290cf33300)
0x21290d005d84  embedded object  (0x10715ed2e019 <String[16]: MAX_POST_LENGTH2>)
0x21290d005d9e  code target (STUB)  (0x21290ce843c0)
0x21290d005da4  code target with id (COMPARE_IC)  (0x21290cea6500) (id=26)
0x21290d005dea  embedded object  (0x10715ed2e9b9 Cell for 6144)
0x21290d005dfa  code target (BUILTIN)  (0x21290cf33260)
0x21290d005e01  embedded object  (0x10715ed2e9b9 Cell for 6144)

--- End code ---
--- Raw source ---
(a) {
    let result = 0;

    if(a > MAX_POST_LENGTH2) {
        result = 1;
    }
    else {
        result = 2;
    }

    return result;
}


--- Code ---
source_position = 259
kind = FUNCTION
name = test2
compiler = full-codegen
Instructions (size = 236)
0x21290d005ec0     0  55             push rbp
0x21290d005ec1     1  4889e5         REX.W movq rbp,rsp
0x21290d005ec4     4  56             push rsi
0x21290d005ec5     5  57             push rdi
0x21290d005ec6     6  488b4f2f       REX.W movq rcx,[rdi+0x2f]
0x21290d005eca    10  488b490f       REX.W movq rcx,[rcx+0xf]
0x21290d005ece    14  83411b01       addl [rcx+0x1b],0x1
0x21290d005ed2    18  41ff75a0       push [r13-0x60]
0x21290d005ed6    22  4d8b55a8       REX.W movq r10,[r13-0x58]
0x21290d005eda    26  4c8955e8       REX.W movq [rbp-0x18],r10
0x21290d005ede    30  493ba5500d0000 REX.W cmpq rsp,[r13+0xd50]
0x21290d005ee5    37  7305           jnc 44  (0x21290d005eec)
0x21290d005ee7    39  e814d4f2ff     call StackCheck  (0x21290cf33300)    ;; code: BUILTIN
0x21290d005eec    44  33c0           xorl rax,rax
0x21290d005eee    46  488945e8       REX.W movq [rbp-0x18],rax
0x21290d005ef2    50  ff7510         push [rbp+0x10]
0x21290d005ef5    53  488bc6         REX.W movq rax,rsi
0x21290d005ef8    56  488b4037       REX.W movq rax,[rax+0x37]
0x21290d005efc    60  493b45a8       REX.W cmpq rax,[r13-0x58]
0x21290d005f00    64  7520           jnz 98  (0x21290d005f22)
0x21290d005f02    66  49ba19e0d25e71100000 REX.W movq r10,0x10715ed2e019    ;; object: 0x10715ed2e019 <String[16]: MAX_POST_LENGTH2>
0x21290d005f0c    76  4152           push r10
0x21290d005f0e    78  b801000000     movl rax,0x1
0x21290d005f13    83  48bbf04bb50000000000 REX.W movq rbx,0xb54bf0
0x21290d005f1d    93  e89ee4e7ff     call 0x21290ce843c0     ;; code: STUB, CEntryStub, minor: 8
0x21290d005f22    98  5a             pop rdx
0x21290d005f23    99  e8d805eaff     call 0x21290cea6500     ;; code: COMPARE_IC UNINITIALIZED (id = 26)
0x21290d005f28   104  90             nop
0x21290d005f29   105  eb0f           jmp 122  (0x21290d005f3a)
0x21290d005f2b   107  493b45b8       REX.W cmpq rax,[r13-0x48]
0x21290d005f2f   111  0f840e000000   jz 131  (0x21290d005f43)
0x21290d005f35   117  e91c000000     jmp 150  (0x21290d005f56)
0x21290d005f3a   122  4885c0         REX.W testq rax,rax
0x21290d005f3d   125  0f8e13000000   jle 150  (0x21290d005f56)
0x21290d005f43   131  48b80000000001000000 REX.W movq rax,0x100000000
0x21290d005f4d   141  488945e8       REX.W movq [rbp-0x18],rax
0x21290d005f51   145  e90e000000     jmp 164  (0x21290d005f64)
0x21290d005f56   150  48b80000000002000000 REX.W movq rax,0x200000000
0x21290d005f60   160  488945e8       REX.W movq [rbp-0x18],rax
0x21290d005f64   164  488b45e8       REX.W movq rax,[rbp-0x18]
0x21290d005f68   168  48bb49ebd25e71100000 REX.W movq rbx,0x10715ed2eb49    ;; object: 0x10715ed2eb49 Cell for 6144
0x21290d005f72   178  83430bd1       addl [rbx+0xb],0xd1
0x21290d005f76   182  791f           jns 215  (0x21290d005f97)
0x21290d005f78   184  50             push rax
0x21290d005f79   185  e8e2d2f2ff     call InterruptCheck  (0x21290cf33260)    ;; code: BUILTIN
0x21290d005f7e   190  58             pop rax
0x21290d005f7f   191  48bb49ebd25e71100000 REX.W movq rbx,0x10715ed2eb49    ;; object: 0x10715ed2eb49 Cell for 6144
0x21290d005f89   201  49ba0000000000180000 REX.W movq r10,0x180000000000
0x21290d005f93   211  4c895307       REX.W movq [rbx+0x7],r10
0x21290d005f97   215  c9             leavel
0x21290d005f98   216  c21000         ret 0x10
0x21290d005f9b   219  498b45a0       REX.W movq rax,[r13-0x60]
0x21290d005f9f   223  e9c4ffffff     jmp 168  (0x21290d005f68)
0x21290d005fa4   228  0f1f4000       nop

Source positions:
 pc offset  position
         0       259
        44       282  statement
        46       282
        50       290  statement
        50       293
        53       297
        98       295
       131       325  statement
       141       332
       150       362  statement
       160       369
       164       385  statement
       164       392
       215       400  statement

Deoptimization Output Data (deopt points = 28)
ast id        pc  state
     2        22  NO_REGISTERS
     3        22  NO_REGISTERS
     4        30  NO_REGISTERS
     8        44  NO_REGISTERS
    18        46  TOS_REGISTER
    13        50  TOS_REGISTER
    11        50  NO_REGISTERS
     9        50  NO_REGISTERS
    29        50  NO_REGISTERS
    27        53  NO_REGISTERS
    32        53  NO_REGISTERS
    30        98  TOS_REGISTER
    24       107  TOS_REGISTER
    22       131  NO_REGISTERS
    33       131  NO_REGISTERS
    43       141  TOS_REGISTER
    38       145  TOS_REGISTER
    36       145  NO_REGISTERS
    34       145  NO_REGISTERS
    23       150  NO_REGISTERS
    46       150  NO_REGISTERS
    56       160  TOS_REGISTER
    51       164  TOS_REGISTER
    49       164  NO_REGISTERS
    47       164  NO_REGISTERS
    21       164  NO_REGISTERS
    61       164  NO_REGISTERS
    59       168  TOS_REGISTER

Back edges (size = 0)
ast_id  pc_offset  loop_depth

0x10715ed2ed49: [TypeFeedbackInfo]
 - ic_total_count: 1, ic_with_type_info_count: 0, ic_generic_count: 0

RelocInfo (size = 10)
0x21290d005ee8  code target (BUILTIN)  (0x21290cf33300)
0x21290d005f04  embedded object  (0x10715ed2e019 <String[16]: MAX_POST_LENGTH2>)
0x21290d005f1e  code target (STUB)  (0x21290ce843c0)
0x21290d005f24  code target with id (COMPARE_IC)  (0x21290cea6500) (id=26)
0x21290d005f6a  embedded object  (0x10715ed2eb49 Cell for 6144)
0x21290d005f7a  code target (BUILTIN)  (0x21290cf33260)
0x21290d005f81  embedded object  (0x10715ed2eb49 Cell for 6144)

--- End code ---
--- Raw source ---
(a) {
    let result = 0;

    if(a > MAX_POST_LENGTH2) {
        result = 1;
    }
    else {
        result = 2;
    }

    return result;
}


--- Optimized code ---
optimization_id = 0
source_position = 259
kind = OPTIMIZED_FUNCTION
name = test2
stack_slots = 5
compiler = crankshaft
Instructions (size = 151)
0x21290d006040     0  55             push rbp
0x21290d006041     1  4889e5         REX.W movq rbp,rsp
0x21290d006044     4  56             push rsi
0x21290d006045     5  57             push rdi
0x21290d006046     6  4883ec08       REX.W subq rsp,0x8
0x21290d00604a    10  488b45f8       REX.W movq rax,[rbp-0x8]
0x21290d00604e    14  488945e8       REX.W movq [rbp-0x18],rax
0x21290d006052    18  488bf0         REX.W movq rsi,rax
0x21290d006055    21  493ba5500d0000 REX.W cmpq rsp,[r13+0xd50]
0x21290d00605c    28  7305           jnc 35  (0x21290d006063)
0x21290d00605e    30  e89dd2f2ff     call StackCheck  (0x21290cf33300)    ;; code: BUILTIN
0x21290d006063    35  488b45e8       REX.W movq rax,[rbp-0x18]
0x21290d006067    39  488b5837       REX.W movq rbx,[rax+0x37]
0x21290d00606b    43  493b5da8       REX.W cmpq rbx,[r13-0x58]
0x21290d00606f    47  0f843f000000   jz 116  (0x21290d0060b4)
0x21290d006075    53  488b5510       REX.W movq rdx,[rbp+0x10]
0x21290d006079    57  f6c201         testb rdx,0x1
0x21290d00607c    60  0f8537000000   jnz 121  (0x21290d0060b9)
0x21290d006082    66  f6c301         testb rbx,0x1
0x21290d006085    69  0f8533000000   jnz 126  (0x21290d0060be)
0x21290d00608b    75  483bd3         REX.W cmpq rdx,rbx
0x21290d00608e    78  0f8f0f000000   jg 99  (0x21290d0060a3)
0x21290d006094    84  48b80000000002000000 REX.W movq rax,0x200000000
0x21290d00609e    94  e90a000000     jmp 109  (0x21290d0060ad)
0x21290d0060a3    99  48b80000000001000000 REX.W movq rax,0x100000000
0x21290d0060ad   109  488be5         REX.W movq rsp,rbp
0x21290d0060b0   112  5d             pop rbp
0x21290d0060b1   113  c21000         ret 0x10
0x21290d0060b4   116  e851dfcfff     call 0x21290cd0400a     ;; deoptimization bailout 1
0x21290d0060b9   121  e856dfcfff     call 0x21290cd04014     ;; deoptimization bailout 2
0x21290d0060be   126  e85bdfcfff     call 0x21290cd0401e     ;; deoptimization bailout 3
0x21290d0060c3   131  90             nop

Source positions:
 pc offset  position
        39       295
        53       295
        57       295
        66       295
        66       295
        75       295
        75       295
        75       295
        84       295
        84       295
        84       295
        94       295
        99       295
        99       295
        99       295
       109       295
       109       295
       109       295
       109       295
       116       295

Inlined functions (count = 0)

Deoptimization Input Data (deopt points = 4)
 index  ast id    argc     pc
     0       4       0     35
     1       4       0     -1
     2       4       0     -1
     3       4       0     -1

Safepoints (size = 19)
0x21290d006063    35  10000 (sp -> fp)       0

RelocInfo (size = 9)
0x21290d00605f  code target (BUILTIN)  (0x21290cf33300)
0x21290d0060b5  runtime entry  (deoptimization bailout 1)
0x21290d0060ba  runtime entry  (deoptimization bailout 2)
0x21290d0060bf  runtime entry  (deoptimization bailout 3)

--- End code ---
--- Raw source ---
(a) {
    let result = 0;

    if(a > MAX_POST_LENGTH1) {
        result = 1;
    }
    else {
        result = 2;
    }

    return result;
}


--- Code ---
source_position = 101
kind = FUNCTION
name = test1
compiler = full-codegen
Instructions (size = 236)
0x21290d006160     0  55             push rbp
0x21290d006161     1  4889e5         REX.W movq rbp,rsp
0x21290d006164     4  56             push rsi
0x21290d006165     5  57             push rdi
0x21290d006166     6  488b4f2f       REX.W movq rcx,[rdi+0x2f]
0x21290d00616a    10  488b490f       REX.W movq rcx,[rcx+0xf]
0x21290d00616e    14  83411b01       addl [rcx+0x1b],0x1
0x21290d006172    18  41ff75a0       push [r13-0x60]
0x21290d006176    22  4d8b55a8       REX.W movq r10,[r13-0x58]
0x21290d00617a    26  4c8955e8       REX.W movq [rbp-0x18],r10
0x21290d00617e    30  493ba5500d0000 REX.W cmpq rsp,[r13+0xd50]
0x21290d006185    37  7305           jnc 44  (0x21290d00618c)
0x21290d006187    39  e874d1f2ff     call StackCheck  (0x21290cf33300)    ;; code: BUILTIN
0x21290d00618c    44  33c0           xorl rax,rax
0x21290d00618e    46  488945e8       REX.W movq [rbp-0x18],rax
0x21290d006192    50  ff7510         push [rbp+0x10]
0x21290d006195    53  488bc6         REX.W movq rax,rsi
0x21290d006198    56  488b402f       REX.W movq rax,[rax+0x2f]
0x21290d00619c    60  493b45a8       REX.W cmpq rax,[r13-0x58]
0x21290d0061a0    64  7520           jnz 98  (0x21290d0061c2)
0x21290d0061a2    66  49baf1dfd25e71100000 REX.W movq r10,0x10715ed2dff1    ;; object: 0x10715ed2dff1 <String[16]: MAX_POST_LENGTH1>
0x21290d0061ac    76  4152           push r10
0x21290d0061ae    78  b801000000     movl rax,0x1
0x21290d0061b3    83  48bbf04bb50000000000 REX.W movq rbx,0xb54bf0
0x21290d0061bd    93  e8fee1e7ff     call 0x21290ce843c0     ;; code: STUB, CEntryStub, minor: 8
0x21290d0061c2    98  5a             pop rdx
0x21290d0061c3    99  e83803eaff     call 0x21290cea6500     ;; code: COMPARE_IC UNINITIALIZED (id = 26)
0x21290d0061c8   104  90             nop
0x21290d0061c9   105  eb0f           jmp 122  (0x21290d0061da)
0x21290d0061cb   107  493b45b8       REX.W cmpq rax,[r13-0x48]
0x21290d0061cf   111  0f840e000000   jz 131  (0x21290d0061e3)
0x21290d0061d5   117  e91c000000     jmp 150  (0x21290d0061f6)
0x21290d0061da   122  4885c0         REX.W testq rax,rax
0x21290d0061dd   125  0f8e13000000   jle 150  (0x21290d0061f6)
0x21290d0061e3   131  48b80000000001000000 REX.W movq rax,0x100000000
0x21290d0061ed   141  488945e8       REX.W movq [rbp-0x18],rax
0x21290d0061f1   145  e90e000000     jmp 164  (0x21290d006204)
0x21290d0061f6   150  48b80000000002000000 REX.W movq rax,0x200000000
0x21290d006200   160  488945e8       REX.W movq [rbp-0x18],rax
0x21290d006204   164  488b45e8       REX.W movq rax,[rbp-0x18]
0x21290d006208   168  48bbc9efd25e71100000 REX.W movq rbx,0x10715ed2efc9    ;; object: 0x10715ed2efc9 Cell for 6144
0x21290d006212   178  83430bd1       addl [rbx+0xb],0xd1
0x21290d006216   182  791f           jns 215  (0x21290d006237)
0x21290d006218   184  50             push rax
0x21290d006219   185  e842d0f2ff     call InterruptCheck  (0x21290cf33260)    ;; code: BUILTIN
0x21290d00621e   190  58             pop rax
0x21290d00621f   191  48bbc9efd25e71100000 REX.W movq rbx,0x10715ed2efc9    ;; object: 0x10715ed2efc9 Cell for 6144
0x21290d006229   201  49ba0000000000180000 REX.W movq r10,0x180000000000
0x21290d006233   211  4c895307       REX.W movq [rbx+0x7],r10
0x21290d006237   215  c9             leavel
0x21290d006238   216  c21000         ret 0x10
0x21290d00623b   219  498b45a0       REX.W movq rax,[r13-0x60]
0x21290d00623f   223  e9c4ffffff     jmp 168  (0x21290d006208)
0x21290d006244   228  0f1f4000       nop

Source positions:
 pc offset  position
         0       101
        44       124  statement
        46       124
        50       132  statement
        50       135
        53       139
        98       137
       131       167  statement
       141       174
       150       204  statement
       160       211
       164       227  statement
       164       234
       215       242  statement

Deoptimization Output Data (deopt points = 28)
ast id        pc  state
     2        22  NO_REGISTERS
     3        22  NO_REGISTERS
     4        30  NO_REGISTERS
     8        44  NO_REGISTERS
    18        46  TOS_REGISTER
    13        50  TOS_REGISTER
    11        50  NO_REGISTERS
     9        50  NO_REGISTERS
    29        50  NO_REGISTERS
    27        53  NO_REGISTERS
    32        53  NO_REGISTERS
    30        98  TOS_REGISTER
    24       107  TOS_REGISTER
    22       131  NO_REGISTERS
    33       131  NO_REGISTERS
    43       141  TOS_REGISTER
    38       145  TOS_REGISTER
    36       145  NO_REGISTERS
    34       145  NO_REGISTERS
    23       150  NO_REGISTERS
    46       150  NO_REGISTERS
    56       160  TOS_REGISTER
    51       164  TOS_REGISTER
    49       164  NO_REGISTERS
    47       164  NO_REGISTERS
    21       164  NO_REGISTERS
    61       164  NO_REGISTERS
    59       168  TOS_REGISTER

Back edges (size = 0)
ast_id  pc_offset  loop_depth

0x10715ed2f1c9: [TypeFeedbackInfo]
 - ic_total_count: 1, ic_with_type_info_count: 0, ic_generic_count: 0

RelocInfo (size = 10)
0x21290d006188  code target (BUILTIN)  (0x21290cf33300)
0x21290d0061a4  embedded object  (0x10715ed2dff1 <String[16]: MAX_POST_LENGTH1>)
0x21290d0061be  code target (STUB)  (0x21290ce843c0)
0x21290d0061c4  code target with id (COMPARE_IC)  (0x21290cea6500) (id=26)
0x21290d00620a  embedded object  (0x10715ed2efc9 Cell for 6144)
0x21290d00621a  code target (BUILTIN)  (0x21290cf33260)
0x21290d006221  embedded object  (0x10715ed2efc9 Cell for 6144)

--- End code ---
--- Raw source ---
(a) {
    let result = 0;

    if(a > MAX_POST_LENGTH1) {
        result = 1;
    }
    else {
        result = 2;
    }

    return result;
}


--- Optimized code ---
optimization_id = 1
source_position = 101
kind = OPTIMIZED_FUNCTION
name = test1
stack_slots = 5
compiler = crankshaft
Instructions (size = 151)
0x21290d0062e0     0  55             push rbp
0x21290d0062e1     1  4889e5         REX.W movq rbp,rsp
0x21290d0062e4     4  56             push rsi
0x21290d0062e5     5  57             push rdi
0x21290d0062e6     6  4883ec08       REX.W subq rsp,0x8
0x21290d0062ea    10  488b45f8       REX.W movq rax,[rbp-0x8]
0x21290d0062ee    14  488945e8       REX.W movq [rbp-0x18],rax
0x21290d0062f2    18  488bf0         REX.W movq rsi,rax
0x21290d0062f5    21  493ba5500d0000 REX.W cmpq rsp,[r13+0xd50]
0x21290d0062fc    28  7305           jnc 35  (0x21290d006303)
0x21290d0062fe    30  e8fdcff2ff     call StackCheck  (0x21290cf33300)    ;; code: BUILTIN
0x21290d006303    35  488b45e8       REX.W movq rax,[rbp-0x18]
0x21290d006307    39  488b582f       REX.W movq rbx,[rax+0x2f]
0x21290d00630b    43  493b5da8       REX.W cmpq rbx,[r13-0x58]
0x21290d00630f    47  0f843f000000   jz 116  (0x21290d006354)
0x21290d006315    53  488b5510       REX.W movq rdx,[rbp+0x10]
0x21290d006319    57  f6c201         testb rdx,0x1
0x21290d00631c    60  0f8537000000   jnz 121  (0x21290d006359)
0x21290d006322    66  f6c301         testb rbx,0x1
0x21290d006325    69  0f8533000000   jnz 126  (0x21290d00635e)
0x21290d00632b    75  483bd3         REX.W cmpq rdx,rbx
0x21290d00632e    78  0f8f0f000000   jg 99  (0x21290d006343)
0x21290d006334    84  48b80000000002000000 REX.W movq rax,0x200000000
0x21290d00633e    94  e90a000000     jmp 109  (0x21290d00634d)
0x21290d006343    99  48b80000000001000000 REX.W movq rax,0x100000000
0x21290d00634d   109  488be5         REX.W movq rsp,rbp
0x21290d006350   112  5d             pop rbp
0x21290d006351   113  c21000         ret 0x10
0x21290d006354   116  e8b1dccfff     call 0x21290cd0400a     ;; deoptimization bailout 1
0x21290d006359   121  e8b6dccfff     call 0x21290cd04014     ;; deoptimization bailout 2
0x21290d00635e   126  e8bbdccfff     call 0x21290cd0401e     ;; deoptimization bailout 3
0x21290d006363   131  90             nop

Source positions:
 pc offset  position
        39       137
        53       137
        57       137
        66       137
        66       137
        75       137
        75       137
        75       137
        84       137
        84       137
        84       137
        94       137
        99       137
        99       137
        99       137
       109       137
       109       137
       109       137
       109       137
       116       137

Inlined functions (count = 0)

Deoptimization Input Data (deopt points = 4)
 index  ast id    argc     pc
     0       4       0     35
     1       4       0     -1
     2       4       0     -1
     3       4       0     -1

Safepoints (size = 19)
0x21290d006303    35  10000 (sp -> fp)       0

RelocInfo (size = 9)
0x21290d0062ff  code target (BUILTIN)  (0x21290cf33300)
0x21290d006355  runtime entry  (deoptimization bailout 1)
0x21290d00635a  runtime entry  (deoptimization bailout 2)
0x21290d00635f  runtime entry  (deoptimization bailout 3)

--- End code ---
--- Raw source ---
const MAX_POST_LENGTH1 = 1024 * 1024 * 0.5;
const MAX_POST_LENGTH2 = 1024 * 1024 / 2;

function test1(a) {
    let result = 0;

    if(a > MAX_POST_LENGTH1) {
        result = 1;
    }
    else {
        result = 2;
    }

    return result;
}

function test2(a) {
    let result = 0;

    if(a > MAX_POST_LENGTH2) {
        result = 1;
    }
    else {
        result = 2;
    }

    return result;
}

while(1) {
    printErr(test1(128));
    printErr(test2(128));
}


--- Code ---
source_position = 0
kind = FUNCTION
compiler = full-codegen
Instructions (size = 792)
0x21290d006400     0  55             push rbp
0x21290d006401     1  4889e5         REX.W movq rbp,rsp
0x21290d006404     4  56             push rsi
0x21290d006405     5  57             push rdi
0x21290d006406     6  488b4f2f       REX.W movq rcx,[rdi+0x2f]
0x21290d00640a    10  488b490f       REX.W movq rcx,[rcx+0xf]
0x21290d00640e    14  83411b01       addl [rcx+0x1b],0x1
0x21290d006412    18  41ff75a0       push [r13-0x60]
0x21290d006416    22  57             push rdi
0x21290d006417    23  49ba09f4d25e71100000 REX.W movq r10,0x10715ed2f409    ;; object: 0x10715ed2f409 <FixedArray[10]>
0x21290d006421    33  4152           push r10
0x21290d006423    35  b802000000     movl rax,0x2
0x21290d006428    40  48bba054b80000000000 REX.W movq rbx,0xb854a0
0x21290d006432    50  e889dfe7ff     call 0x21290ce843c0     ;; code: STUB, CEntryStub, minor: 8
0x21290d006437    55  488bf0         REX.W movq rsi,rax
0x21290d00643a    58  488945f8       REX.W movq [rbp-0x8],rax
0x21290d00643e    62  4d8b55a8       REX.W movq r10,[r13-0x58]
0x21290d006442    66  4c89562f       REX.W movq [rsi+0x2f],r10
0x21290d006446    70  4d8b55a8       REX.W movq r10,[r13-0x58]
0x21290d00644a    74  4c895637       REX.W movq [rsi+0x37],r10
0x21290d00644e    78  49ba79f4d25e71100000 REX.W movq r10,0x10715ed2f479    ;; object: 0x10715ed2f479 <FixedArray[6]>
0x21290d006458    88  4152           push r10
0x21290d00645a    90  6a00           push 0x0
0x21290d00645c    92  488b45f0       REX.W movq rax,[rbp-0x10]
0x21290d006460    96  488b402f       REX.W movq rax,[rax+0x2f]
0x21290d006464   100  488b400f       REX.W movq rax,[rax+0xf]
0x21290d006468   104  50             push rax
0x21290d006469   105  b803000000     movl rax,0x3
0x21290d00646e   110  48bb60f0b70000000000 REX.W movq rbx,0xb7f060
0x21290d006478   120  e843dfe7ff     call 0x21290ce843c0     ;; code: STUB, CEntryStub, minor: 8
0x21290d00647d   125  493ba5500d0000 REX.W cmpq rsp,[r13+0xd50]
0x21290d006484   132  7305           jnc 139  (0x21290d00648b)
0x21290d006486   134  e875cef2ff     call StackCheck  (0x21290cf33300)    ;; code: BUILTIN
0x21290d00648b   139  48b800000000b8487639 REX.W movq rax,0x397648b800000000
0x21290d006495   149  49ba00000000b8487e39 REX.W movq r10,0x397e48b800000000
0x21290d00649f   159  4933c2         REX.W xorq rax,r10
0x21290d0064a2   162  488bce         REX.W movq rcx,rsi
0x21290d0064a5   165  4889412f       REX.W movq [rcx+0x2f],rax
0x21290d0064a9   169  488bd0         REX.W movq rdx,rax
0x21290d0064ac   172  f6c201         testb rdx,0x1
0x21290d0064af   175  0f8426000000   jz 219  (0x21290d0064db)
0x21290d0064b5   181  488d592f       REX.W leaq rbx,[rcx+0x2f]
0x21290d0064b9   185  4881e20000f8ff REX.W andq rdx,0xfffffffffff80000
0x21290d0064c0   192  f6420802       testb [rdx+0x8],0x2
0x21290d0064c4   196  7415           jz 219  (0x21290d0064db)
0x21290d0064c6   198  48c7c20000f8ff REX.W movq rdx,0xfff80000
0x21290d0064cd   205  4823d1         REX.W andq rdx,rcx
0x21290d0064d0   208  f6420804       testb [rdx+0x8],0x4
0x21290d0064d4   212  7405           jz 219  (0x21290d0064db)
0x21290d0064d6   214  e86536eaff     call 0x21290cea9b40     ;; code: STUB, RecordWriteStub, minor: 801
0x21290d0064db   219  48b800000000b8487639 REX.W movq rax,0x397648b800000000
0x21290d0064e5   229  49ba00000000b8487e39 REX.W movq r10,0x397e48b800000000
0x21290d0064ef   239  4933c2         REX.W xorq rax,r10
0x21290d0064f2   242  488bce         REX.W movq rcx,rsi
0x21290d0064f5   245  48894137       REX.W movq [rcx+0x37],rax
0x21290d0064f9   249  488bd0         REX.W movq rdx,rax
0x21290d0064fc   252  f6c201         testb rdx,0x1
0x21290d0064ff   255  0f8426000000   jz 299  (0x21290d00652b)
0x21290d006505   261  488d5937       REX.W leaq rbx,[rcx+0x37]
0x21290d006509   265  4881e20000f8ff REX.W andq rdx,0xfffffffffff80000
0x21290d006510   272  f6420802       testb [rdx+0x8],0x2
0x21290d006514   276  7415           jz 299  (0x21290d00652b)
0x21290d006516   278  48c7c20000f8ff REX.W movq rdx,0xfff80000
0x21290d00651d   285  4823d1         REX.W andq rdx,rcx
0x21290d006520   288  f6420804       testb [rdx+0x8],0x4
0x21290d006524   292  7405           jz 299  (0x21290d00652b)
0x21290d006526   294  e81536eaff     call 0x21290cea9b40     ;; code: STUB, RecordWriteStub, minor: 801
0x21290d00652b   299  48b8112350e3ec240000 REX.W movq rax,0x24ece3502311    ;; object: 0x24ece3502311 <undefined>
0x21290d006535   309  488945e8       REX.W movq [rbp-0x18],rax
0x21290d006539   313  eb0f           jmp 330  (0x21290d00654a)
0x21290d00653b   315  493b45b8       REX.W cmpq rax,[r13-0x48]
0x21290d00653f   319  0f8405000000   jz 330  (0x21290d00654a)
0x21290d006545   325  e97a010000     jmp 708  (0x21290d0066c4)
0x21290d00654a   330  48b96177d25e71100000 REX.W movq rcx,0x10715ed27761    ;; object: 0x10715ed27761 <String[8]: printErr>
0x21290d006554   340  48b8000000000a000000 REX.W movq rax,0xa00000000
0x21290d00655e   350  e83d5af4ff     call LoadGlobalICTrampoline  (0x21290cf4bfa0)    ;; code: LOAD_GLOBAL_IC
0x21290d006563   355  488b75f8       REX.W movq rsi,[rbp-0x8]
0x21290d006567   359  50             push rax
0x21290d006568   360  49ba112350e3ec240000 REX.W movq r10,0x24ece3502311    ;; object: 0x24ece3502311 <undefined>
0x21290d006572   370  4152           push r10
0x21290d006574   372  48b941e0d25e71100000 REX.W movq rcx,0x10715ed2e041    ;; object: 0x10715ed2e041 <String[5]: test1>
0x21290d00657e   382  48b80000000002000000 REX.W movq rax,0x200000000
0x21290d006588   392  e8135af4ff     call LoadGlobalICTrampoline  (0x21290cf4bfa0)    ;; code: LOAD_GLOBAL_IC
0x21290d00658d   397  488b75f8       REX.W movq rsi,[rbp-0x8]
0x21290d006591   401  50             push rax
0x21290d006592   402  49ba112350e3ec240000 REX.W movq r10,0x24ece3502311    ;; object: 0x24ece3502311 <undefined>
0x21290d00659c   412  4152           push r10
0x21290d00659e   414  49ba0000000080000000 REX.W movq r10,0x8000000000
0x21290d0065a8   424  4152           push r10
0x21290d0065aa   426  48ba000000000c000000 REX.W movq rdx,0xc00000000
0x21290d0065b4   436  488b7c2410     REX.W movq rdi,[rsp+0x10]
0x21290d0065b9   441  b801000000     movl rax,0x1
0x21290d0065be   446  e8bdf1ffff     call 0x21290d005780     ;; code: CALL_IC
0x21290d0065c3   451  488b75f8       REX.W movq rsi,[rbp-0x8]
0x21290d0065c7   455  48890424       REX.W movq [rsp],rax
0x21290d0065cb   459  48ba0000000008000000 REX.W movq rdx,0x800000000
0x21290d0065d5   469  488b7c2410     REX.W movq rdi,[rsp+0x10]
0x21290d0065da   474  b801000000     movl rax,0x1
0x21290d0065df   479  e89cf1ffff     call 0x21290d005780     ;; code: CALL_IC
0x21290d0065e4   484  488b75f8       REX.W movq rsi,[rbp-0x8]
0x21290d0065e8   488  4883c408       REX.W addq rsp,0x8
0x21290d0065ec   492  48b96177d25e71100000 REX.W movq rcx,0x10715ed27761    ;; object: 0x10715ed27761 <String[8]: printErr>
0x21290d0065f6   502  48b8000000000a000000 REX.W movq rax,0xa00000000
0x21290d006600   512  e89b59f4ff     call LoadGlobalICTrampoline  (0x21290cf4bfa0)    ;; code: LOAD_GLOBAL_IC
0x21290d006605   517  488b75f8       REX.W movq rsi,[rbp-0x8]
0x21290d006609   521  50             push rax
0x21290d00660a   522  49ba112350e3ec240000 REX.W movq r10,0x24ece3502311    ;; object: 0x24ece3502311 <undefined>
0x21290d006614   532  4152           push r10
0x21290d006616   534  48b961e0d25e71100000 REX.W movq rcx,0x10715ed2e061    ;; object: 0x10715ed2e061 <String[5]: test2>
0x21290d006620   544  48b80000000005000000 REX.W movq rax,0x500000000
0x21290d00662a   554  e87159f4ff     call LoadGlobalICTrampoline  (0x21290cf4bfa0)    ;; code: LOAD_GLOBAL_IC
0x21290d00662f   559  488b75f8       REX.W movq rsi,[rbp-0x8]
0x21290d006633   563  50             push rax
0x21290d006634   564  49ba112350e3ec240000 REX.W movq r10,0x24ece3502311    ;; object: 0x24ece3502311 <undefined>
0x21290d00663e   574  4152           push r10
0x21290d006640   576  49ba0000000080000000 REX.W movq r10,0x8000000000
0x21290d00664a   586  4152           push r10
0x21290d00664c   588  48ba0000000010000000 REX.W movq rdx,0x1000000000
0x21290d006656   598  488b7c2410     REX.W movq rdi,[rsp+0x10]
0x21290d00665b   603  b801000000     movl rax,0x1
0x21290d006660   608  e81bf1ffff     call 0x21290d005780     ;; code: CALL_IC
0x21290d006665   613  488b75f8       REX.W movq rsi,[rbp-0x8]
0x21290d006669   617  48890424       REX.W movq [rsp],rax
0x21290d00666d   621  48ba000000000e000000 REX.W movq rdx,0xe00000000
0x21290d006677   631  488b7c2410     REX.W movq rdi,[rsp+0x10]
0x21290d00667c   636  b801000000     movl rax,0x1
0x21290d006681   641  e8faf0ffff     call 0x21290d005780     ;; code: CALL_IC
0x21290d006686   646  488b75f8       REX.W movq rsi,[rbp-0x8]
0x21290d00668a   650  4883c408       REX.W addq rsp,0x8
0x21290d00668e   654  488945e8       REX.W movq [rbp-0x18],rax
0x21290d006692   658  48bb69f4d25e71100000 REX.W movq rbx,0x10715ed2f469    ;; object: 0x10715ed2f469 Cell for 6144
0x21290d00669c   668  83430bfe       addl [rbx+0xb],0xfe
0x21290d0066a0   672  791d           jns 703  (0x21290d0066bf)
0x21290d0066a2   674  e8b9cbf2ff     call InterruptCheck  (0x21290cf33260)    ;; code: BUILTIN
0x21290d0066a7   679  48bb69f4d25e71100000 REX.W movq rbx,0x10715ed2f469    ;; object: 0x10715ed2f469 Cell for 6144
0x21290d0066b1   689  49ba0000000000180000 REX.W movq r10,0x180000000000
0x21290d0066bb   699  4c895307       REX.W movq [rbx+0x7],r10
0x21290d0066bf   703  e975feffff     jmp 313  (0x21290d006539)
0x21290d0066c4   708  488b45e8       REX.W movq rax,[rbp-0x18]
0x21290d0066c8   712  48bb69f4d25e71100000 REX.W movq rbx,0x10715ed2f469    ;; object: 0x10715ed2f469 Cell for 6144
0x21290d0066d2   722  83430bfc       addl [rbx+0xb],0xfc
0x21290d0066d6   726  791f           jns 759  (0x21290d0066f7)
0x21290d0066d8   728  50             push rax
0x21290d0066d9   729  e882cbf2ff     call InterruptCheck  (0x21290cf33260)    ;; code: BUILTIN
0x21290d0066de   734  58             pop rax
0x21290d0066df   735  48bb69f4d25e71100000 REX.W movq rbx,0x10715ed2f469    ;; object: 0x10715ed2f469 Cell for 6144
0x21290d0066e9   745  49ba0000000000180000 REX.W movq r10,0x180000000000
0x21290d0066f3   755  4c895307       REX.W movq [rbx+0x7],r10
0x21290d0066f7   759  c9             leavel
0x21290d0066f8   760  c20800         ret 0x8
0x21290d0066fb   763  498b45a0       REX.W movq rax,[r13-0x60]
0x21290d0066ff   767  e9c4ffffff     jmp 712  (0x21290d0066c8)
0x21290d006704   772  0f1f4000       nop

Source positions:
 pc offset  position
         0         0
       139        37  statement
       162        37
       219        81  statement
       242        81
       313       409  statement
       330       418  statement
       330       418
       372       427
       426       427
       459       418
       492       444  statement
       492       444
       534       453
       588       453
       621       444
       759       467  statement

Deoptimization Output Data (deopt points = 54)
ast id        pc  state
     1        55  TOS_REGISTER
     2        62  NO_REGISTERS
     3        62  NO_REGISTERS
    28        70  NO_REGISTERS
    41        78  NO_REGISTERS
     4       125  NO_REGISTERS
    21       139  NO_REGISTERS
    31       162  TOS_REGISTER
    26       219  TOS_REGISTER
    24       219  NO_REGISTERS
    22       219  NO_REGISTERS
    34       219  NO_REGISTERS
    44       242  TOS_REGISTER
    39       299  TOS_REGISTER
    37       299  NO_REGISTERS
    35       299  NO_REGISTERS
    47       299  NO_REGISTERS
    57       309  TOS_REGISTER
    52       313  TOS_REGISTER
    50       313  NO_REGISTERS
    64       315  TOS_REGISTER
    63       330  NO_REGISTERS
    67       330  NO_REGISTERS
    76       330  NO_REGISTERS
    74       360  NO_REGISTERS
    83       372  NO_REGISTERS
    81       402  NO_REGISTERS
    84       426  NO_REGISTERS
    80       426  NO_REGISTERS
    79       451  TOS_REGISTER
    77       459  NO_REGISTERS
    73       459  NO_REGISTERS
    72       484  TOS_REGISTER
    70       492  NO_REGISTERS
   100       492  NO_REGISTERS
    98       522  NO_REGISTERS
   107       534  NO_REGISTERS
   105       564  NO_REGISTERS
   108       588  NO_REGISTERS
   104       588  NO_REGISTERS
   103       613  TOS_REGISTER
   101       621  NO_REGISTERS
    97       621  NO_REGISTERS
    96       646  TOS_REGISTER
    94       654  TOS_REGISTER
    89       658  TOS_REGISTER
    87       658  NO_REGISTERS
    68       658  NO_REGISTERS
    60       703  NO_REGISTERS
    62       703  NO_REGISTERS
    61       708  NO_REGISTERS
    48       708  NO_REGISTERS
   113       708  NO_REGISTERS
   111       712  TOS_REGISTER

Back edges (size = 1)
ast_id  pc_offset  loop_depth
    62        679           1

0x10715ed2f861: [TypeFeedbackInfo]
 - ic_total_count: 4, ic_with_type_info_count: 0, ic_generic_count: 0

RelocInfo (size = 34)
0x21290d006419  embedded object  (0x10715ed2f409 <FixedArray[10]>)
0x21290d006433  code target (STUB)  (0x21290ce843c0)
0x21290d006450  embedded object  (0x10715ed2f479 <FixedArray[6]>)
0x21290d006479  code target (STUB)  (0x21290ce843c0)
0x21290d006487  code target (BUILTIN)  (0x21290cf33300)
0x21290d0064d7  code target (STUB)  (0x21290cea9b40)
0x21290d006527  code target (STUB)  (0x21290cea9b40)
0x21290d00652d  embedded object  (0x24ece3502311 <undefined>)
0x21290d00654c  embedded object  (0x10715ed27761 <String[8]: printErr>)
0x21290d00655f  code target (LOAD_GLOBAL_IC)  (0x21290cf4bfa0)
0x21290d00656a  embedded object  (0x24ece3502311 <undefined>)
0x21290d006576  embedded object  (0x10715ed2e041 <String[5]: test1>)
0x21290d006589  code target (LOAD_GLOBAL_IC)  (0x21290cf4bfa0)
0x21290d006594  embedded object  (0x24ece3502311 <undefined>)
0x21290d0065bf  code target (CALL_IC)  (0x21290d005780)
0x21290d0065e0  code target (CALL_IC)  (0x21290d005780)
0x21290d0065ee  embedded object  (0x10715ed27761 <String[8]: printErr>)
0x21290d006601  code target (LOAD_GLOBAL_IC)  (0x21290cf4bfa0)
0x21290d00660c  embedded object  (0x24ece3502311 <undefined>)
0x21290d006618  embedded object  (0x10715ed2e061 <String[5]: test2>)
0x21290d00662b  code target (LOAD_GLOBAL_IC)  (0x21290cf4bfa0)
0x21290d006636  embedded object  (0x24ece3502311 <undefined>)
0x21290d006661  code target (CALL_IC)  (0x21290d005780)
0x21290d006682  code target (CALL_IC)  (0x21290d005780)
0x21290d006694  embedded object  (0x10715ed2f469 Cell for 6144)
0x21290d0066a3  code target (BUILTIN)  (0x21290cf33260)
0x21290d0066a9  embedded object  (0x10715ed2f469 Cell for 6144)
0x21290d0066ca  embedded object  (0x10715ed2f469 Cell for 6144)
0x21290d0066da  code target (BUILTIN)  (0x21290cf33260)
0x21290d0066e1  embedded object  (0x10715ed2f469 Cell for 6144)

--- End code ---
--- Raw source ---
const MAX_POST_LENGTH1 = 1024 * 1024 * 0.5;
const MAX_POST_LENGTH2 = 1024 * 1024 / 2;

function test1(a) {
    let result = 0;

    if(a > MAX_POST_LENGTH1) {
        result = 1;
    }
    else {
        result = 2;
    }

    return result;
}

function test2(a) {
    let result = 0;

    if(a > MAX_POST_LENGTH2) {
        result = 1;
    }
    else {
        result = 2;
    }

    return result;
}

while(1) {
    printErr(test1(128));
    printErr(test2(128));
}


--- Optimized code ---
optimization_id = 2
source_position = 0
kind = OPTIMIZED_FUNCTION
stack_slots = 8
compiler = crankshaft
Instructions (size = 902)
0x21290d007240     0  55             push rbp
0x21290d007241     1  4889e5         REX.W movq rbp,rsp
0x21290d007244     4  56             push rsi
0x21290d007245     5  57             push rdi
0x21290d007246     6  4883ec20       REX.W subq rsp,0x20
0x21290d00724a    10  0f1f00         nop
0x21290d00724d    13  57             push rdi
0x21290d00724e    14  49ba09f4d25e71100000 REX.W movq r10,0x10715ed2f409    ;; object: 0x10715ed2f409 <FixedArray[10]>
0x21290d007258    24  4152           push r10
0x21290d00725a    26  b802000000     movl rax,0x2
0x21290d00725f    31  498d9d58b25dfe REX.W leaq rbx,[r13-0x1a24da8]
0x21290d007266    38  e855d1e7ff     call 0x21290ce843c0     ;; code: STUB, CEntryStub, minor: 8
0x21290d00726b    43  488bf0         REX.W movq rsi,rax
0x21290d00726e    46  488945f8       REX.W movq [rbp-0x8],rax
0x21290d007272    50  488b45f8       REX.W movq rax,[rbp-0x8]
0x21290d007276    54  488945e0       REX.W movq [rbp-0x20],rax
0x21290d00727a    58  48bb512350e3ec240000 REX.W movq rbx,0x24ece3502351    ;; object: 0x24ece3502351 <the hole>
0x21290d007284    68  4889582f       REX.W movq [rax+0x2f],rbx
0x21290d007288    72  48bb512350e3ec240000 REX.W movq rbx,0x24ece3502351    ;; object: 0x24ece3502351 <the hole>
0x21290d007292    82  48895837       REX.W movq [rax+0x37],rbx
0x21290d007296    86  488bf0         REX.W movq rsi,rax
0x21290d007299    89  49bab9f8d25e71100000 REX.W movq r10,0x10715ed2f8b9    ;; object: 0x10715ed2f8b9 <FixedArray[6]>
0x21290d0072a3    99  4152           push r10
0x21290d0072a5   101  6a00           push 0x0
0x21290d0072a7   103  49ba19e5d25e71100000 REX.W movq r10,0x10715ed2e519    ;; object: 0x10715ed2e519 <FixedArray[18]>
0x21290d0072b1   113  4152           push r10
0x21290d0072b3   115  b803000000     movl rax,0x3
0x21290d0072b8   120  498d9d184e5dfe REX.W leaq rbx,[r13-0x1a2b1e8]
0x21290d0072bf   127  e8fcd0e7ff     call 0x21290ce843c0     ;; code: STUB, CEntryStub, minor: 8
0x21290d0072c4   132  488b75e0       REX.W movq rsi,[rbp-0x20]
0x21290d0072c8   136  660f1f840000000000 nop
0x21290d0072d1   145  493ba5500d0000 REX.W cmpq rsp,[r13+0xd50]
0x21290d0072d8   152  7305           jnc 159  (0x21290d0072df)
0x21290d0072da   154  e821c0f2ff     call StackCheck  (0x21290cf33300)    ;; code: BUILTIN
0x21290d0072df   159  48b80000000000000800 REX.W movq rax,0x8000000000000
0x21290d0072e9   169  488b75e0       REX.W movq rsi,[rbp-0x20]
0x21290d0072ed   173  4889462f       REX.W movq [rsi+0x2f],rax
0x21290d0072f1   177  a801           test al,0x1
0x21290d0072f3   179  0f8425000000   jz 222  (0x21290d00731e)
0x21290d0072f9   185  488d5e2f       REX.W leaq rbx,[rsi+0x2f]
0x21290d0072fd   189  48250000f8ff   REX.W and rax,0xfffffffffff80000
0x21290d007303   195  f6400802       testb [rax+0x8],0x2
0x21290d007307   199  7415           jz 222  (0x21290d00731e)
0x21290d007309   201  48c7c00000f8ff REX.W movq rax,0xfff80000
0x21290d007310   208  4823c6         REX.W andq rax,rsi
0x21290d007313   211  f6400804       testb [rax+0x8],0x4
0x21290d007317   215  7405           jz 222  (0x21290d00731e)
0x21290d007319   217  e862f6ffff     call 0x21290d006980     ;; code: STUB, RecordWriteStub, minor: 8966
0x21290d00731e   222  48b80000000000000800 REX.W movq rax,0x8000000000000
0x21290d007328   232  48894637       REX.W movq [rsi+0x37],rax
0x21290d00732c   236  a801           test al,0x1
0x21290d00732e   238  0f8425000000   jz 281  (0x21290d007359)
0x21290d007334   244  488d5e37       REX.W leaq rbx,[rsi+0x37]
0x21290d007338   248  48250000f8ff   REX.W and rax,0xfffffffffff80000
0x21290d00733e   254  f6400802       testb [rax+0x8],0x2
0x21290d007342   258  7415           jz 281  (0x21290d007359)
0x21290d007344   260  48c7c00000f8ff REX.W movq rax,0xfff80000
0x21290d00734b   267  4823c6         REX.W andq rax,rsi
0x21290d00734e   270  f6400804       testb [rax+0x8],0x4
0x21290d007352   274  7405           jz 281  (0x21290d007359)
0x21290d007354   276  e827f6ffff     call 0x21290d006980     ;; code: STUB, RecordWriteStub, minor: 8966
0x21290d007359   281  e918000000     jmp 310  (0x21290d007376)
0x21290d00735e   286  4883ec18       REX.W subq rsp,0x18
0x21290d007362   290  488b75f8       REX.W movq rsi,[rbp-0x8]
0x21290d007366   294  4c8b4d10       REX.W movq r9,[rbp+0x10]
0x21290d00736a   298  4c8bc6         REX.W movq r8,rsi
0x21290d00736d   301  488b45e8       REX.W movq rax,[rbp-0x18]
0x21290d007371   305  e911000000     jmp 327  (0x21290d007387)
0x21290d007376   310  4c8b4d10       REX.W movq r9,[rbp+0x10]
0x21290d00737a   314  4c8bc6         REX.W movq r8,rsi
0x21290d00737d   317  48b8112350e3ec240000 REX.W movq rax,0x24ece3502311    ;; object: 0x24ece3502311 <undefined>
0x21290d007387   327  4c894dd8       REX.W movq [rbp-0x28],r9
0x21290d00738b   331  4c8945e0       REX.W movq [rbp-0x20],r8
0x21290d00738f   335  4d8d9df8fee5fd REX.W leaq r11,[r13-0x21a0108]
0x21290d007396   342  4c895dd0       REX.W movq [rbp-0x30],r11
0x21290d00739a   346  493ba5500d0000 REX.W cmpq rsp,[r13+0xd50]
0x21290d0073a1   353  0f820f010000   jc 630  (0x21290d0074b6)
0x21290d0073a7   359  48b811e6d25e71100000 REX.W movq rax,0x10715ed2e611    ;; object: 0x10715ed2e611 <FixedArray[6]>
0x21290d0073b1   369  488b402f       REX.W movq rax,[rax+0x2f]
0x21290d0073b5   373  493b45a8       REX.W cmpq rax,[r13-0x58]
0x21290d0073b9   377  0f84a6010000   jz 805  (0x21290d007565)
0x21290d0073bf   383  a801           test al,0x1
0x21290d0073c1   385  0f8534010000   jnz 699  (0x21290d0074fb)
0x21290d0073c7   391  48c1e820       REX.W shrq rax, 32
0x21290d0073cb   395  3d80000000     cmp rax,0x80
0x21290d0073d0   400  0f8c0f000000   jl 421  (0x21290d0073e5)
0x21290d0073d6   406  48b80000000002000000 REX.W movq rax,0x200000000
0x21290d0073e0   416  e90a000000     jmp 431  (0x21290d0073ef)
0x21290d0073e5   421  48b80000000001000000 REX.W movq rax,0x100000000
0x21290d0073ef   431  49ba413278e0e8060000 REX.W movq r10,0x6e8e0783241    ;; object: 0x6e8e0783241 <JS Global Object>
0x21290d0073f9   441  4152           push r10
0x21290d0073fb   443  50             push rax
0x21290d0073fc   444  48bfd976d25e71100000 REX.W movq rdi,0x10715ed276d9    ;; object: 0x10715ed276d9 <JS Function printErr (SharedFunctionInfo 0x10715ed27621)>
0x21290d007406   454  48bb112350e3ec240000 REX.W movq rbx,0x24ece3502311    ;; object: 0x24ece3502311 <undefined>
0x21290d007410   464  48b9413278e0e8060000 REX.W movq rcx,0x6e8e0783241    ;; object: 0x6e8e0783241 <JS Global Object>
0x21290d00741a   474  498bf0         REX.W movq rsi,r8
0x21290d00741d   477  498bd3         REX.W movq rdx,r11
0x21290d007420   480  e87bf3ffff     call 0x21290d0067a0     ;; code: STUB, CallApiCallbackStub, minor: 6
0x21290d007425   485  48b811e6d25e71100000 REX.W movq rax,0x10715ed2e611    ;; object: 0x10715ed2e611 <FixedArray[6]>
0x21290d00742f   495  488b4037       REX.W movq rax,[rax+0x37]
0x21290d007433   499  493b45a8       REX.W cmpq rax,[r13-0x58]
0x21290d007437   503  0f842d010000   jz 810  (0x21290d00756a)
0x21290d00743d   509  a801           test al,0x1
0x21290d00743f   511  0f85eb000000   jnz 752  (0x21290d007530)
0x21290d007445   517  48c1e820       REX.W shrq rax, 32
0x21290d007449   521  3d80000000     cmp rax,0x80
0x21290d00744e   526  0f8c0f000000   jl 547  (0x21290d007463)
0x21290d007454   532  48b80000000002000000 REX.W movq rax,0x200000000
0x21290d00745e   542  e90a000000     jmp 557  (0x21290d00746d)
0x21290d007463   547  48b80000000001000000 REX.W movq rax,0x100000000
0x21290d00746d   557  49ba413278e0e8060000 REX.W movq r10,0x6e8e0783241    ;; object: 0x6e8e0783241 <JS Global Object>
0x21290d007477   567  4152           push r10
0x21290d007479   569  50             push rax
0x21290d00747a   570  48bfd976d25e71100000 REX.W movq rdi,0x10715ed276d9    ;; object: 0x10715ed276d9 <JS Function printErr (SharedFunctionInfo 0x10715ed27621)>
0x21290d007484   580  48bb112350e3ec240000 REX.W movq rbx,0x24ece3502311    ;; object: 0x24ece3502311 <undefined>
0x21290d00748e   590  48b9413278e0e8060000 REX.W movq rcx,0x6e8e0783241    ;; object: 0x6e8e0783241 <JS Global Object>
0x21290d007498   600  488b75e0       REX.W mov
```